var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

```diff
@@ -72,7 +73,9 @@ class PyPennylaneLightningKokkos(CMakePackage, PythonExtension, CudaPackage, ROC
     depends_on("py-pybind11", type="link")
     depends_on("py-pip", type="build")
     depends_on("py-wheel", type="build")
-    depends_on("py-pennylane@0.28:", type=("build", "run"))
+    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@:0.31.0")
```

Comment:
```suggestion
    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@0.31:")
```
This will also apply to the main branch